elkodon/src/port/subscriber.rs

```diff
@@ -139,10 +140,15 @@ impl<'a, 'config: 'a, Service: service::Details<'config>, MessageType: Debug>
         Ok(new_self)
     }
 
-    pub(crate) fn release_sample(&self, channel_id: usize, message: *const MessageType) {
+    pub(crate) fn release_sample<Header: Debug>(
```

Comment:
When I looked at the code I wondered. Are types that implement `Drop` compatible with shm ipc communication and our interface?
I think they are since `assume_init` converts the sample into an initialized one. When a samples memory is released/deallocated, is drop called when it is initialized.

I apologize for these thoughts or sin from the past but could you please check it out since you are already working at the right position. If this does not work could you please create an issue.

Reply:
As long as it is a `MaybeUninit<T>` drop is not called on `T` but after `assume_init` is called, the compiler threats it as a `T` and also calls drop on it.

I think currently types with `Drop` are compatible with shm communication but I did not look at the implementation of the queue overflow handling. It depends on the availability of the type information at that location and if `drop` is called there.

With the language bindings this might need to be restricted to types not implementing drop, though.